lib/PuppeteerSharp/Helpers/TempDirectory.cs

```diff
@@ -29,37 +29,44 @@ private TempDirectory(string path)
             Path = path;
         }
 
-        ~TempDirectory()
-        {
-            DisposeCore();
-        }
+        ~TempDirectory() => Dispose();
 
         public string Path { get; }
 
-        public void Dispose()
+#pragma warning disable CA1816
+        public void Dispose() => _ = DisposeAsync();
+#pragma warning restore CA1816
```

Comment:
We could delete `Dispose` and `IDisposable`.
This helps ensuring we call and await `DeleteAsync`, except if `~TempDirectory()` is called.

Reply:
rollback